### PR TITLE
Added PYPI

### PR DESCRIPTION
**Pull request by mvalliammal**
# Description

Done all the needed for the addition of PYPI

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Python Version(s) Tested:
* Tenable.sc version (if necessary):

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -113,8 +138,12 @@ try {
     }
 
     parallel(tasks)
-
     common.setResultIfNotSet(Constants.JSUCCESS)
+    
+    if (env.BRANCH_NAME == 'master' && releaseBuild == 'Yes') {
```
Comment (rboergertenable) [inline Jenkinsfile:143]: This variable is no longer defined. Please make this a build parameter.

Comment (rboergertenable) [reply Jenkinsfile:143]: Example here: 

https://github.eng.tenable.com/Product/kafka-connect-configurator/blob/32cf403cc6153cfcbe3e7e8c25c9c5b776d0a6cc/Jenkinsfile#L10

File: Jenkinsfile [hunk 2]
```diff
@@ -5,6 +5,18 @@ import com.tenable.jenkins.common.*
 import com.tenable.jenkins.Constants
 import com.tenable.jenkins.builds.snyk.*
 import com.tenable.jenkins.builds.nexusiq.*
+
```
Comment (rboergertenable) [inline Jenkinsfile:8]: Please remove the unneeded blank lines in the Jenkinsfile.

File: Jenkinsfile [hunk 3]
```diff
@@ -5,6 +5,18 @@ import com.tenable.jenkins.common.*
 import com.tenable.jenkins.Constants
 import com.tenable.jenkins.builds.snyk.*
 import com.tenable.jenkins.builds.nexusiq.*
+    
+
+
+def addParameters(final script, def theParams) {
+
```
Comment (rboergertenable) [inline Jenkinsfile:12]: This parameter should only be added on master.

i.e.

if (script.env.BRANCH_NAME == 'master') {
....

File: Jenkinsfile [hunk 4]
```diff
@@ -5,6 +5,18 @@ import com.tenable.jenkins.common.*
...
+
+def addParameters(final script, def theParams) {
+    
+    theParams << script.booleanParam(defaultValue: false,
+        description: 'Puplish to PYPI!', name: 'PUPBLISH2PYPI')
+
+    theParams
+}
+PropertiesHelper.defaultCICDProperties(this, 0, this.&addParameters)
```
Comment (rboergertenable) [inline Jenkinsfile:18]: We should disable the params that we don't need:

i.e. 

replace this, 0 

with

this, Constants.PROP_NOCONSEC | Constants.PROP_NOSCHEDULE

File: Jenkinsfile [hunk 5]
```diff
@@ -6,8 +6,16 @@ import com.tenable.jenkins.Constants
 import com.tenable.jenkins.builds.snyk.*
 import com.tenable.jenkins.builds.nexusiq.*
 
-pythonVersion = [ '3.6', '3.7', '3.8', '3.9' ]
+if (env.BRANCH_NAME == 'master') {
```
Comment (rboergertenable) [inline Jenkinsfile:9]: Please move inside the function. witt script.env.

File: Jenkinsfile [hunk 6]
```diff
@@ -6,8 +6,17 @@ import com.tenable.jenkins.Constants
 import com.tenable.jenkins.builds.snyk.*
 import com.tenable.jenkins.builds.nexusiq.*
 
+def addParameters(final script, def theParams) {
+    if (script.env.BRANCH_NAME == 'master') {
+        theParams << script.booleanParam(defaultValue: false,
+            description: 'Puplish to PYPI!', name: 'PUPBLISH2PYPI')
+        theParams
```
Comment (rboergertenable) [inline Jenkinsfile:13]: Please move below the closing bracket of the if

File: Jenkinsfile [hunk 7]
```diff
@@ -6,8 +6,16 @@ import com.tenable.jenkins.Constants
 import com.tenable.jenkins.builds.snyk.*
 import com.tenable.jenkins.builds.nexusiq.*
 
+def addParameters(final script, def theParams) {
+    if (script.env.BRANCH_NAME == 'master') {
+       theParams << script.booleanParam(defaultValue: false, description: 'Puplish to PYPI!', name: 'PUPBLISH2PYPI')
+    }
+    theParams
+}
+if (env.BRANCH_NAME == 'MASTER') {
```
Comment (rboergertenable) [inline Jenkinsfile:15]: Please remove this. The build parameters are there to allow QG overwrites. They are valid on feature branches as well.

**Review by rboergertenable — CHANGES_REQUESTED** [review]

**Review by SteveMcGrath — COMMENTED** [review]
Rejecting as Jenkinsfile will not be used

**Pull request closed without merging**